src/BugsnagUnity/Native/Android/NativePayloadClassWrapper.cs

```diff
@@ -22,5 +23,183 @@ public void SetNativeString(string key, string value)
         {
             NativePointer.Call(key, value);
         }
+
+        public int? GetNativeInt(string key)
+        {
+            var number = NativePointer.Call<AndroidJavaObject>(key);
+            if (number == null)
+            {
+                return null;
+            }
+            return number.Call<int>("intValue");
+        }
+
+        public void SetNativeInt(string key, int? value)
+        {
+            if (value == null)
+            {
+                NativePointer.Call(key, null);
+            }
+            else
+            {
+                var javaInteger = new AndroidJavaObject("java.lang.Integer", (int)value);
+                NativePointer.Call(key, javaInteger);
+            }            
+        }
+
+        public void SetNativeLong(string key, int? value)
+        {
+            if (value == null)
+            {
+                NativePointer.Call(key, null);
+            }
+            else
+            {
+                var javaInteger = new AndroidJavaObject("java.lang.Long", (long)value);
+                NativePointer.Call(key, javaInteger);
+            }
+        }
+
+        public bool? GetNativeBool(string key)
+        {
+            var value = NativePointer.Call<AndroidJavaObject>(key);
+            if (value == null)
+            {
+                return null;
+            }
+            else
+            {
+                return value.Call<bool>("booleanValue");
+            }
+        }
+
+        public void SetNativeBool(string key, bool? value)
+        {
+            if (value == null)
+            {
+                NativePointer.Call(key, null);
+            }
+            else
+            {
+                var javaBoolean = new AndroidJavaObject("java.lang.Boolean", (bool)value);
+                NativePointer.Call(key, javaBoolean);
+            }
+        }
+
+        public string[] GetNativeStringArray(string key)
```

Comment:
It might be worth returning `List<string>` here (or another method that does the equivalent). I'm wary that this might lead to a situation where a collection is editable on Android, but remains a static size in C#

Reply:
Another method for mutable collections is coming in the next PR